internal/slicer/slicer.go

```diff
@@ -170,9 +172,18 @@ func Run(options *RunOptions) (*Report, error) {
 				if extractInfo == nil {
 					return nil
 				}
-				if _, ok := slice.Contents[extractInfo.Path]; !ok {
+				if _, ok := pathInfos[extractInfo.Path]; !ok {
 					return nil
 				}
+				for _, s := range pkgSlices[slice.Package] {
```

Comment:
What is the difference between pkgSlices[slice.Package] and slice.Package.Slices?

Why do we need this in addition to the condition above?

We already had a bug in this logic, and it still feels like we're a bit trying to see "what sticks".

Reply:
`release.Packages[slice.Package].Slices` contains all the list of slices from the release and we want only those which are also in the selection. That is the difference.

Reply:
This is needed because we iterate over `options.Selection.Slices` to preserve the selection order and mark the packages as used on first occurrence. Because in the end we iterate over packages, we need a map to slices.

Another option is to create a more complex data structure that has pkg and slices in the order of the selection (sort of an array of arrays). That is also doable.

Reply:
The way this is constructing an extraction list and then attempting to find out what matched out of the breadcrumps left behind doesn't feel entirely sound. I may be wrong, but this seems error prone still.  For example, what happens if two different slices on the same package glob a given file through different expressions?  Which of them will we get here, and then how do we map one slice to the other?

Reply:
Those are indeed very good questions. I think one of the problems for this bug is that we do not cover testing several slices for the same package at all.

Even if it is not strictly related with this bug, I will add the necessary tests and then we'll see if chisel as a whole works properly or fix it. From preliminary extra tests it seems like the latter :smiling_face_with_tear: .